File: java/src/test/java/io/binac/leetcode/MaximumProductSubarrayTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MaximumProductSubarrayTest {
    private static final MaximumProductSubarray.Solution1 SOLUTION1 = new MaximumProductSubarray.Solution1();

    private final int[] NUMS1 = {2, 3, -2, 4};
    private final int EXPECTED1 = 6;

    private final int[] NUMS2 = {-2, 0, -1};
    private final int EXPECTED2 = 0;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.maxProduct(NUMS1));
        assertEquals(EXPECTED2, SOLUTION1.maxProduct(NUMS2));
    }
}

File: cpp/unittests/MaximumProductSubarrayTest.cpp
#include "MaximumProductSubarray.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(MaximumProductSubarray, Solution152_1) {
  std::vector<int> Nums1{2, 3, -2, 4};
  const int Expected1 = 6;
  EXPECT_EQ(Expected1, Solution152_1().maxProduct(Nums1));

  std::vector<int> Nums2{-2, 0, -1};
  const int Expected2 = 0;
  EXPECT_EQ(Expected2, Solution152_1().maxProduct(Nums2));
}